### PR TITLE
Add circular convolution

### PR DESCRIPTION
**Pull request by subhendudash02**
### Describe your change:

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms include at least one URL that points to Wikipedia or another similar explanation.
* [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: electronics/circular_convolution.py [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+        >>> convolution.first_signal = [1, 0.5]
+        >>> convolution.second_signal = [0.5, 1]
+        >>> convolution.circular_convolution()
+        [1.0, 1.25]
+
+        >>> convolution.first_signal = [0.2, 0.4, 0.6, 0.8, 1.0, 1.2, 1.4, 1.6]
+        >>> convolution.second_signal = [0.1, 0.3, 0.5, 0.7, 0.9, 1.1, 1.3, 1.5]
+        >>> convolution.circular_convolution()
+        [5.2, 6.0, 6.48, 6.64, 6.48, 6.0, 5.2, 4.08]
+
```
Comment (Abinashbunty) [inline electronics/circular_convolution.py:54]: Add another unit test case where `first_signal` & `second_signal` have unequal sizes.

Comment (subhendudash02) [reply electronics/circular_convolution.py:54]: Thanks for reviewing!
I will add the test case for unequal signals soon!

**Review by Abinashbunty — APPROVED** [review]
LGTM @cclauss

**Comment by subhendudash02** [comment]
@cclauss please check whether this PR is valid.

File: electronics/circular_convolution.py [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+        max_length = max(length_first_signal, length_second_signal)
+
+        # create a zero matrix of max_length x max_length
+        matrix = [[0] * max_length for i in range(max_length)]
+
+        # fills the smaller signal with zeros to make both signals of same length
+        if length_first_signal < length_second_signal:
+            self.first_signal += [0 for i in range(max_length - length_first_signal)]
+        elif length_first_signal > length_second_signal:
+            self.second_signal += [0 for i in range(max_length - length_second_signal)]
```
Comment (cclauss) [inline electronics/circular_convolution.py:74]:
```suggestion
            self.first_signal += [[0] * (max_length - length_first_signal)]
        elif length_first_signal > length_second_signal:
            self.second_signal += [[0] * (max_length - length_second_signal)]
```

File: electronics/circular_convolution.py [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+            [x[1], x[0], x[3], x[2]],
+            [x[2], x[1], x[0], x[3]],
+            [x[3], x[2], x[1], x[0]]
+        ]
+        """
+        for i in range(max_length):
+            rotated_signal = deque(self.second_signal)
+            rotated_signal.rotate(i)
+            for j in range(max_length):
+                matrix[i][j] += rotated_signal[j]
```
Comment (cclauss) [inline electronics/circular_convolution.py:89]:
```suggestion
            for j, item in enumerate(rotated_signal):
                matrix[i][j] += item
```

File: electronics/circular_convolution.py [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+
+        # multiply the matrix with the first signal
+        final_signal = np.matmul(np.transpose(matrix), np.transpose(self.first_signal))
+
+        # rounding-off to two decimal places
+        final_signal = [round(i, 2) for i in final_signal]
+
+        return final_signal
```
Comment (cclauss) [inline electronics/circular_convolution.py:97]:
```suggestion
        return [round(i, 2) for i in final_signal]
```

**Review by cclauss — CHANGES_REQUESTED** [review]
Looking good...

**Comment by subhendudash02** [comment]
@cclauss modified the changes you suggested!

**Review by cclauss — APPROVED** [review]

**Pull request merged**